### PR TITLE
High: minio: Resource agent for the minio storage server

### PR DESCRIPTION
**Pull request by krig**
Minio is a distributed object storage server
built for cloud applications and devops.

URL: https://www.minio.io/

Fixes: #1012

### CODE REVIEW

File: heartbeat/minio [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+</longdesc>
+<shortdesc lang="en">OCF Resource Agent compliant Minio server script.</shortdesc>
+
+
+<parameters>
+
+<parameter name="binary" unique="0" required="0">
+<longdesc lang="en">The Minio server binary</longdesc>
+<shortdesc lang="en">The Minio server binary</shortdesc>
+<content type="string" default="/opt/minio/minio-server" />
```
Comment (oalbrigt) [inline heartbeat/minio]: These should be defined as _default variables.

File: heartbeat/minio [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+
+if [ $# -ne 1 ]
+then
+  usage
+  exit $OCF_ERR_ARGS
+fi
+
+case $1 in
+    start)      minio_validate_all
+                        minio_start
```
Comment (oalbrigt) [inline heartbeat/minio]: Cleanup in indentation here would help making this case-statement easier to read.

**Comment by krig** [comment]
@oalbrigt @tsmgeek Added some cleanup commits.

**Comment by oalbrigt** [comment]
The cleanups looks good to me, but it's still failing CI in regards to metadata it seems.

**Comment by krig** [comment]
Ah, the CI setup is completely broken ATM due to changes at Travis. I've looked at fixing it, but it's complicated :( No solution yet...

**Comment by krig** [comment]
Ah but you are right, the agent is broken as well.. I'll fix it.

**Pull request merged**